finesse/gui/interferometer_monitor.py

```diff
@@ -0,0 +1,204 @@
+"""Panel and widgets related to monitoring the interferometer."""
+from copy import deepcopy
+from typing import Optional
+
+from PySide6.QtCore import Qt
+from PySide6.QtWidgets import QGridLayout, QGroupBox, QLabel, QLineEdit
+
+from .led_icons import LEDIcon
+
+
+def get_vals_from_server() -> dict:
+    """Placeholder function for retrieving interferometer properties.
+
+    Returns:
+        data_table: A dictionary containing the physical properties being monitored
+    """
+    data_table = {
+        "PSF27 Temp": [28.151062, "deg. C"],
+        "Cryo Temp": [0.0, "deg. K"],
+        "Blackbody Hum": [2.463968, "%"],
+        "Source Temp": [70.007156, "deg. C"],
+        "Aux Volt": [6.285875, "V"],
+        "Aux Curr": [0.910230, "A"],
+        "Laser Curr": [0.583892, "A"],
+    }
+    return data_table
+
+
+class EM27Monitor(QGroupBox):
+    """Widgets to view the EM27 properties."""
+
+    def __init__(self, prop_labels: list[str], prop_units: list[Optional[str]]) -> None:
+        """Creates a sequence of widgets to view properties monitored by the EM27."""
+        super().__init__("EM27 SOH Monitor")
+
+        self._prop_labels = deepcopy(prop_labels)
+        self._prop_units = deepcopy(prop_units)
+
+        self._psf27_temp_box = QLineEdit(self._prop_units[0])
+        self._psf27_temp_box.setReadOnly(True)
+        self._psf27_temp_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
```

Comment:
All these controls could be added dynamically in a `for` loop. The problem you have at the mo is the code makes assumptions that all the values will be present and in a specific order, which isn't very robust.